### DISCUSSION TITLE
RavenDB Group By with max()/min()

### DISCUSSION THREAD

**Discussion by jeremy-morren** [ravendb/ravendb#13917] [Q&A] [answered]
I have a collection with the structure `{ Name, Timestamp, Value}` where Value is an ever-increasing column. I need to run a query to find the difference between the largest and the smallest value of `Value` between arbitrary timestamps, grouped by `Name`.  In a SQL database, the query

``` SQL
SELECT [Name], MAX([Value]) - MIN([Value]) as [Difference]
FROM [Collection]
WHERE [Timestamp] BETWEEN @from AND @to
GROUP BY [Name]
```

would suffice.  I am a newcomer to RavenDB, but I cannot seem to convert the above query to RQL.

**Comment by ayende** [comment, marked as answer]
Isn't that just a time series setup? In that case, the query would be something like this:

See: 

https://ravendb.net/docs/article-page/5.0/csharp/studio/database/document-extensions/time-series

Using the sample data, you can run this query:


```
from "Employees" 
where id() = 'employees/1-A'
select timeseries(
    from Heartrates
    select min(), max()    
), FirstName
```
